{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\r\n",
      "echo \" Pipeline stated: $(date)\"\r\n",
      "makeblastdb -in diatoms.sequences.FINAL2017.fasta -out diatoms -dbtype nucl\r\n",
      "echo \" QC started: $(date)\"\r\n",
      "python ./ampliconQC.py --data $1 --forward ATGCGTTGGAGAGARCGTTTC --reverse GATCACCTTCTAATTTACCWACAACTG --threads 130 --histograms --qiime\r\n",
      "echo \" QIIME Pipeline: $(date)\"\r\n",
      "for file in $1/*.passedQC.fastq; do awk 'NR%4==2{sum+=1}END{print FILENAME,sum}' $file >> $1/diatomSequenceCounts.txt; done\r\n",
      "echo $(date)\r\n",
      "pick_otus.py -i $1/readyForQiime.allsamples.fasta -o $1/picked_otus_97 \r\n",
      "echo $(date)\r\n",
      "pick_rep_set.py -i $1/picked_otus_97/readyForQiime.allsamples_otus.txt -f $1/readyForQiime.allsamples.fasta -o $1/repset.fasta\r\n",
      "echo $(date)\r\n",
      "blastn -db diatoms -query $1/repset.fasta -out $1/repset.diatoms.blastn -task blastn -max_target_seqs 1 -num_threads 130 -outfmt 6 -evalue 0.01\r\n",
      "echo $(date)\r\n",
      "mkdir $1/assigned_taxonomy\r\n",
      "echo $(date)\r\n",
      "python ./create_taxonomy_assignments_from_blast.py --taxonomy diatoms.taxonomy.FINAL2017.txt --percid 95.0 --blast $1/repset.diatoms.blastn --output $1/assigned_taxonomy/repset.taxonomy.txt\r\n",
      "echo $(date)\r\n",
      "make_otu_table.py -i $1/picked_otus_97/readyForQiime.allsamples_otus.txt -t $1/assigned_taxonomy/repset.taxonomy.txt -o $1/otu_table.biom\r\n",
      "echo \" Lookup taxa from OTU table: $(date)\"\r\n",
      "filter_taxa_from_otu_table.py -i $1/otu_table.biom -o $1/otu_table.diatomsonly.biom -n MARINE,NOT_DIATOM,Yellow_green_Algae,None\r\n",
      "echo \" Sort OTU table: $(date)\"\r\n",
      "sort_otu_table.py -i $1/otu_table.diatomsonly.biom -o $1/otu_table.diatomsonly.biom\r\n",
      "echo \" Summerise taxa: $(date)\"\r\n",
      "summarize_taxa.py -L 1 -i $1/otu_table.diatomsonly.biom -o $1/visualised_taxonomy -a\r\n",
      "echo \" Produce Diatom reports: $(date)\"\r\n",
      "python ./produceDiatomReports.py --folder $1 --lookup $2\r\n",
      "echo \"Pipeline finished: $(date)\"\r\n"
     ]
    }
   ],
   "source": [
    "!cat diatomPipeline.dms"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create reference database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 05/31/2019 14:11:32\n",
      "New DB name:   /code/diatoms\n",
      "New DB title:  diatoms.sequences.FINAL2017.fasta\n",
      "Sequence type: Nucleotide\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 2701 sequences in 0.189506 seconds.\n"
     ]
    }
   ],
   "source": [
    "!makeblastdb -in diatoms.sequences.FINAL2017.fasta -out diatoms -dbtype nucl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Run quality control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 1.9.1 with Python 2.7.16\r\n",
      "Command line parameters: -e 0.047619047619 -b ATGCGTTGGAGAGARCGTTTC -b GATCACCTTCTAATTTACCWACAACTG -o /code/sequences/101.R2.fastq.gz.trimmed.fastq.gz /code/sequences/101.R2.fastq.gz\r\n",
      "Trimming 2 adapters with at most 4.8% errors in single-end mode ...\r\n",
      "Finished in 5.53 s (55 us/read; 1.10 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total reads processed:                 101,443\r\n",
      "Reads with adapters:                    98,773 (97.4%)\r\n",
      "Reads written (passing filters):       101,443 (100.0%)\r\n",
      "\r\n",
      "Total basepairs processed:    30,534,343 bp\r\n",
      "Total written (filtered):     27,867,887 bp (91.3%)\r\n",
      "\r\n",
      "=== Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: ATGCGTTGGAGAGARCGTTTC; Type: variable 5'/3'; Length: 21; Trimmed: 19 times.\r\n",
      "2 times, it overlapped the 5' end of a read\r\n",
      "17 times, it overlapped the 3' end or was within the read\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-21 bp: 0\r\n",
      "\r\n",
      "Overview of removed sequences (5')\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "5\t1\t99.1\t0\t1\r\n",
      "8\t1\t1.5\t0\t1\r\n",
      "\r\n",
      "\r\n",
      "Overview of removed sequences (3' or within)\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t4\t1585.0\t0\t4\r\n",
      "4\t12\t396.3\t0\t12\r\n",
      "5\t1\t99.1\t0\t1\r\n",
      "\r\n",
      "=== Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: GATCACCTTCTAATTTACCWACAACTG; Type: variable 5'/3'; Length: 27; Trimmed: 98754 times.\r\n",
      "98711 times, it overlapped the 5' end of a read\r\n",
      "43 times, it overlapped the 3' end or was within the read\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-20 bp: 0; 21-27 bp: 1\r\n",
      "\r\n",
      "Overview of removed sequences (5')\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t1\t1585.0\t0\t1\r\n",
      "4\t2\t396.3\t0\t2\r\n",
      "18\t1\t0.0\t0\t1\r\n",
      "20\t1\t0.0\t0\t1\r\n",
      "21\t6\t0.0\t0\t6\r\n",
      "22\t3\t0.0\t1\t2 1\r\n",
      "23\t1\t0.0\t1\t0 1\r\n",
      "24\t17\t0.0\t1\t9 8\r\n",
      "25\t14\t0.0\t1\t4 10\r\n",
      "26\t2056\t0.0\t1\t78 1978\r\n",
      "27\t96568\t0.0\t1\t90557 6011\r\n",
      "28\t41\t0.0\t1\t0 41\r\n",
      "\r\n",
      "\r\n",
      "Overview of removed sequences (3' or within)\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t21\t1585.0\t0\t21\r\n",
      "4\t4\t396.3\t0\t4\r\n",
      "5\t6\t99.1\t0\t6\r\n",
      "7\t1\t6.2\t0\t1\r\n",
      "300\t11\t0.0\t1\t5 6\r\n",
      "\r\n",
      "This is cutadapt 1.9.1 with Python 2.7.16\r\n",
      "Command line parameters: -e 0.047619047619 -b ATGCGTTGGAGAGARCGTTTC -b GATCACCTTCTAATTTACCWACAACTG -o /code/sequences/101.R1.fastq.gz.trimmed.fastq.gz /code/sequences/101.R1.fastq.gz\r\n",
      "Trimming 2 adapters with at most 4.8% errors in single-end mode ...\r\n",
      "Finished in 5.98 s (59 us/read; 1.02 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total reads processed:                 101,443\r\n",
      "Reads with adapters:                    93,905 (92.6%)\r\n",
      "Reads written (passing filters):       101,443 (100.0%)\r\n",
      "\r\n",
      "Total basepairs processed:    30,534,343 bp\r\n",
      "Total written (filtered):     28,564,203 bp (93.5%)\r\n",
      "\r\n",
      "=== Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: ATGCGTTGGAGAGARCGTTTC; Type: variable 5'/3'; Length: 21; Trimmed: 93833 times.\r\n",
      "93712 times, it overlapped the 5' end of a read\r\n",
      "121 times, it overlapped the 3' end or was within the read\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-21 bp: 0\r\n",
      "\r\n",
      "Overview of removed sequences (5')\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t1\t1585.0\t0\t1\r\n",
      "12\t1\t0.0\t0\t1\r\n",
      "13\t1\t0.0\t0\t1\r\n",
      "16\t1\t0.0\t0\t1\r\n",
      "17\t3\t0.0\t0\t3\r\n",
      "18\t16\t0.0\t0\t16\r\n",
      "19\t8\t0.0\t0\t8\r\n",
      "20\t78\t0.0\t0\t78\r\n",
      "21\t93603\t0.0\t0\t93603\r\n",
      "\r\n",
      "\r\n",
      "Overview of removed sequences (3' or within)\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t111\t1585.0\t0\t111\r\n",
      "4\t2\t396.3\t0\t2\r\n",
      "7\t2\t6.2\t0\t2\r\n",
      "300\t6\t0.0\t0\t6\r\n",
      "\r\n",
      "=== Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: GATCACCTTCTAATTTACCWACAACTG; Type: variable 5'/3'; Length: 27; Trimmed: 72 times.\r\n",
      "20 times, it overlapped the 5' end of a read\r\n",
      "52 times, it overlapped the 3' end or was within the read\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-20 bp: 0; 21-27 bp: 1\r\n",
      "\r\n",
      "Overview of removed sequences (5')\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t20\t1585.0\t0\t20\r\n",
      "\r\n",
      "\r\n",
      "Overview of removed sequences (3' or within)\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t46\t1585.0\t0\t46\r\n",
      "4\t1\t396.3\t0\t1\r\n",
      "5\t5\t99.1\t0\t5\r\n",
      "\r\n",
      "This is cutadapt 1.9.1 with Python 2.7.16\r\n",
      "Command line parameters: -e 0.047619047619 -b ATGCGTTGGAGAGARCGTTTC -b GATCACCTTCTAATTTACCWACAACTG -o /code/sequences/100.R2.fastq.gz.trimmed.fastq.gz /code/sequences/100.R2.fastq.gz\r\n",
      "Trimming 2 adapters with at most 4.8% errors in single-end mode ...\r\n",
      "Finished in 7.30 s (53 us/read; 1.13 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total reads processed:                 138,072\r\n",
      "Reads with adapters:                   135,433 (98.1%)\r\n",
      "Reads written (passing filters):       138,072 (100.0%)\r\n",
      "\r\n",
      "Total basepairs processed:    41,559,672 bp\r\n",
      "Total written (filtered):     37,901,550 bp (91.2%)\r\n",
      "\r\n",
      "=== Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: ATGCGTTGGAGAGARCGTTTC; Type: variable 5'/3'; Length: 21; Trimmed: 17 times.\r\n",
      "2 times, it overlapped the 5' end of a read\r\n",
      "15 times, it overlapped the 3' end or was within the read\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-21 bp: 0\r\n",
      "\r\n",
      "Overview of removed sequences (5')\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t1\t2157.4\t0\t1\r\n",
      "4\t1\t539.3\t0\t1\r\n",
      "\r\n",
      "\r\n",
      "Overview of removed sequences (3' or within)\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t7\t2157.4\t0\t7\r\n",
      "4\t6\t539.3\t0\t6\r\n",
      "5\t2\t134.8\t0\t2\r\n",
      "\r\n",
      "=== Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: GATCACCTTCTAATTTACCWACAACTG; Type: variable 5'/3'; Length: 27; Trimmed: 135416 times.\r\n",
      "135378 times, it overlapped the 5' end of a read\r\n",
      "38 times, it overlapped the 3' end or was within the read\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-20 bp: 0; 21-27 bp: 1\r\n",
      "\r\n",
      "Overview of removed sequences (5')\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "4\t1\t539.3\t0\t1\r\n",
      "5\t1\t134.8\t0\t1\r\n",
      "16\t1\t0.0\t0\t1\r\n",
      "18\t1\t0.0\t0\t1\r\n",
      "21\t4\t0.0\t0\t4\r\n",
      "22\t2\t0.0\t1\t2\r\n",
      "23\t17\t0.0\t1\t7 10\r\n",
      "24\t17\t0.0\t1\t3 14\r\n",
      "25\t11\t0.0\t1\t5 6\r\n",
      "26\t2426\t0.0\t1\t96 2330\r\n",
      "27\t132853\t0.0\t1\t127216 5637\r\n",
      "28\t44\t0.0\t1\t0 44\r\n",
      "\r\n",
      "\r\n",
      "Overview of removed sequences (3' or within)\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t12\t2157.4\t0\t12\r\n",
      "4\t1\t539.3\t0\t1\r\n",
      "5\t6\t134.8\t0\t6\r\n",
      "6\t1\t33.7\t0\t1\r\n",
      "300\t18\t0.0\t1\t14 4\r\n",
      "\r\n",
      "This is cutadapt 1.9.1 with Python 2.7.16\r\n",
      "Command line parameters: -e 0.047619047619 -b ATGCGTTGGAGAGARCGTTTC -b GATCACCTTCTAATTTACCWACAACTG -o /code/sequences/100.R1.fastq.gz.trimmed.fastq.gz /code/sequences/100.R1.fastq.gz\r\n",
      "Trimming 2 adapters with at most 4.8% errors in single-end mode ...\r\n",
      "Finished in 7.66 s (56 us/read; 1.08 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total reads processed:                 138,072\r\n",
      "Reads with adapters:                   131,199 (95.0%)\r\n",
      "Reads written (passing filters):       138,072 (100.0%)\r\n",
      "\r\n",
      "Total basepairs processed:    41,559,672 bp\r\n",
      "Total written (filtered):     38,806,625 bp (93.4%)\r\n",
      "\r\n",
      "=== Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: ATGCGTTGGAGAGARCGTTTC; Type: variable 5'/3'; Length: 21; Trimmed: 131109 times.\r\n",
      "130993 times, it overlapped the 5' end of a read\r\n",
      "116 times, it overlapped the 3' end or was within the read\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-21 bp: 0\r\n",
      "\r\n",
      "Overview of removed sequences (5')\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t1\t2157.4\t0\t1\r\n",
      "4\t1\t539.3\t0\t1\r\n",
      "15\t1\t0.0\t0\t1\r\n",
      "16\t3\t0.0\t0\t3\r\n",
      "17\t1\t0.0\t0\t1\r\n",
      "18\t17\t0.0\t0\t17\r\n",
      "19\t16\t0.0\t0\t16\r\n",
      "20\t71\t0.0\t0\t71\r\n",
      "21\t130882\t0.0\t0\t130882\r\n",
      "\r\n",
      "\r\n",
      "Overview of removed sequences (3' or within)\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t106\t2157.4\t0\t106\r\n",
      "4\t3\t539.3\t0\t3\r\n",
      "5\t1\t134.8\t0\t1\r\n",
      "300\t6\t0.0\t0\t6\r\n",
      "\r\n",
      "=== Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: GATCACCTTCTAATTTACCWACAACTG; Type: variable 5'/3'; Length: 27; Trimmed: 90 times.\r\n",
      "68 times, it overlapped the 5' end of a read\r\n",
      "22 times, it overlapped the 3' end or was within the read\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-20 bp: 0; 21-27 bp: 1\r\n",
      "\r\n",
      "Overview of removed sequences (5')\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t68\t2157.4\t0\t68\r\n",
      "\r\n",
      "\r\n",
      "Overview of removed sequences (3' or within)\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t20\t2157.4\t0\t20\r\n",
      "4\t1\t539.3\t0\t1\r\n",
      "5\t1\t134.8\t0\t1\r\n",
      "\r\n",
      "This is cutadapt 1.9.1 with Python 2.7.16\r\n",
      "Command line parameters: -e 0.047619047619 -b ATGCGTTGGAGAGARCGTTTC -b GATCACCTTCTAATTTACCWACAACTG -o /code/sequences/102.R2.fastq.gz.trimmed.fastq.gz /code/sequences/102.R2.fastq.gz\r\n",
      "Trimming 2 adapters with at most 4.8% errors in single-end mode ...\r\n",
      "Finished in 8.81 s (50 us/read; 1.21 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total reads processed:                 177,704\r\n",
      "Reads with adapters:                   173,011 (97.4%)\r\n",
      "Reads written (passing filters):       177,704 (100.0%)\r\n",
      "\r\n",
      "Total basepairs processed:    53,488,904 bp\r\n",
      "Total written (filtered):     48,814,967 bp (91.3%)\r\n",
      "\r\n",
      "=== Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: ATGCGTTGGAGAGARCGTTTC; Type: variable 5'/3'; Length: 21; Trimmed: 43 times.\r\n",
      "0 times, it overlapped the 5' end of a read\r\n",
      "43 times, it overlapped the 3' end or was within the read\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-21 bp: 0\r\n",
      "\r\n",
      "Overview of removed sequences (5')\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "\r\n",
      "\r\n",
      "Overview of removed sequences (3' or within)\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t28\t2776.6\t0\t28\r\n",
      "4\t13\t694.2\t0\t13\r\n",
      "5\t2\t173.5\t0\t2\r\n",
      "\r\n",
      "=== Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: GATCACCTTCTAATTTACCWACAACTG; Type: variable 5'/3'; Length: 27; Trimmed: 172968 times.\r\n",
      "172888 times, it overlapped the 5' end of a read\r\n",
      "80 times, it overlapped the 3' end or was within the read\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-20 bp: 0; 21-27 bp: 1\r\n",
      "\r\n",
      "Overview of removed sequences (5')\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "18\t1\t0.0\t0\t1\r\n",
      "19\t1\t0.0\t0\t1\r\n",
      "20\t1\t0.0\t0\t1\r\n",
      "21\t7\t0.0\t0\t7\r\n",
      "22\t3\t0.0\t1\t1 2\r\n",
      "23\t12\t0.0\t1\t4 8\r\n",
      "24\t29\t0.0\t1\t20 9\r\n",
      "25\t23\t0.0\t1\t11 12\r\n",
      "26\t3427\t0.0\t1\t111 3316\r\n",
      "27\t169338\t0.0\t1\t153245 16093\r\n",
      "28\t46\t0.0\t1\t0 46\r\n",
      "\r\n",
      "\r\n",
      "Overview of removed sequences (3' or within)\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t31\t2776.6\t0\t31\r\n",
      "4\t14\t694.2\t0\t14\r\n",
      "5\t3\t173.5\t0\t3\r\n",
      "6\t1\t43.4\t0\t1\r\n",
      "294\t2\t0.0\t1\t2\r\n",
      "300\t29\t0.0\t1\t18 11\r\n",
      "\r\n",
      "This is cutadapt 1.9.1 with Python 2.7.16\r\n",
      "Command line parameters: -e 0.047619047619 -b ATGCGTTGGAGAGARCGTTTC -b GATCACCTTCTAATTTACCWACAACTG -o /code/sequences/102.R1.fastq.gz.trimmed.fastq.gz /code/sequences/102.R1.fastq.gz\r\n",
      "Trimming 2 adapters with at most 4.8% errors in single-end mode ...\r\n",
      "Finished in 9.36 s (53 us/read; 1.14 M reads/minute).\r\n",
      "\r\n",
      "=== Summary ===\r\n",
      "\r\n",
      "Total reads processed:                 177,704\r\n",
      "Reads with adapters:                   165,868 (93.3%)\r\n",
      "Reads written (passing filters):       177,704 (100.0%)\r\n",
      "\r\n",
      "Total basepairs processed:    53,488,904 bp\r\n",
      "Total written (filtered):     50,008,252 bp (93.5%)\r\n",
      "\r\n",
      "=== Adapter 1 ===\r\n",
      "\r\n",
      "Sequence: ATGCGTTGGAGAGARCGTTTC; Type: variable 5'/3'; Length: 21; Trimmed: 165771 times.\r\n",
      "165605 times, it overlapped the 5' end of a read\r\n",
      "166 times, it overlapped the 3' end or was within the read\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-21 bp: 0\r\n",
      "\r\n",
      "Overview of removed sequences (5')\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "13\t3\t0.0\t0\t3\r\n",
      "15\t1\t0.0\t0\t1\r\n",
      "16\t5\t0.0\t0\t5\r\n",
      "17\t2\t0.0\t0\t2\r\n",
      "18\t19\t0.0\t0\t19\r\n",
      "19\t24\t0.0\t0\t24\r\n",
      "20\t83\t0.0\t0\t83\r\n",
      "21\t165468\t0.0\t0\t165468\r\n",
      "\r\n",
      "\r\n",
      "Overview of removed sequences (3' or within)\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t150\t2776.6\t0\t150\r\n",
      "4\t6\t694.2\t0\t6\r\n",
      "7\t1\t10.8\t0\t1\r\n",
      "10\t1\t0.2\t0\t1\r\n",
      "300\t8\t0.0\t0\t8\r\n",
      "\r\n",
      "=== Adapter 2 ===\r\n",
      "\r\n",
      "Sequence: GATCACCTTCTAATTTACCWACAACTG; Type: variable 5'/3'; Length: 27; Trimmed: 97 times.\r\n",
      "41 times, it overlapped the 5' end of a read\r\n",
      "56 times, it overlapped the 3' end or was within the read\r\n",
      "\r\n",
      "No. of allowed errors:\r\n",
      "0-20 bp: 0; 21-27 bp: 1\r\n",
      "\r\n",
      "Overview of removed sequences (5')\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t41\t2776.6\t0\t41\r\n",
      "\r\n",
      "\r\n",
      "Overview of removed sequences (3' or within)\r\n",
      "length\tcount\texpect\tmax.err\terror counts\r\n",
      "3\t48\t2776.6\t0\t48\r\n",
      "5\t8\t173.5\t0\t8\r\n",
      "\r\n",
      "\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "FastQ paired records kept: 352174 (176087 pairs)\n",
      "FastQ single records kept: 1532 (from PE1: 1433, from PE2: 99)\n",
      "FastQ paired records discarded: 170 (85 pairs)\n",
      "FastQ single records discarded: 1532 (from PE1: 99, from PE2: 1433)\n",
      "\n",
      "\n",
      "\n",
      "FastQ paired records kept: 272720 (136360 pairs)\n",
      "FastQ single records kept: 1498 (from PE1: 1385, from PE2: 113)\n",
      "FastQ paired records discarded: 428 (214 pairs)\n",
      "FastQ single records discarded: 1498 (from PE1: 113, from PE2: 1385)\n",
      "\n",
      "\n",
      "\n",
      "FastQ paired records kept: 200810 (100405 pairs)\n",
      "FastQ single records kept: 988 (from PE1: 926, from PE2: 62)\n",
      "FastQ paired records discarded: 100 (50 pairs)\n",
      "FastQ single records discarded: 988 (from PE1: 62, from PE2: 926)\n",
      "\n",
      "\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.11 [Nov 5, 2017]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /code/sequences/102.sickle.trimmed.R1.fastq.gz\n",
      "Reverse reads file.................: /code/sequences/102.sickle.trimmed.R2.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: NO\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 130\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Assemblying reads: 0%\n",
      "Assemblying reads: 49%\n",
      "Assemblying reads: 92%\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 166,309 / 176,087 (94.447%)\n",
      "Discarded reads ...................: 0 / 176,087 (0.000%)\n",
      "Not assembled reads ...............: 9,778 / 176,087 (5.553%)\n",
      "Assembled reads file...............: /code/sequences/102.assembled.fastq\n",
      "Discarded reads file...............: /code/sequences/102.discarded.fastq\n",
      "Unassembled forward reads file.....: /code/sequences/102.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /code/sequences/102.unassembled.reverse.fastq\n",
      "\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.11 [Nov 5, 2017]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /code/sequences/100.sickle.trimmed.R1.fastq.gz\n",
      "Reverse reads file.................: /code/sequences/100.sickle.trimmed.R2.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: NO\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 130\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Assemblying reads: 0%\n",
      "Assemblying reads: 63%\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 130,401 / 136,360 (95.630%)\n",
      "Discarded reads ...................: 0 / 136,360 (0.000%)\n",
      "Not assembled reads ...............: 5,959 / 136,360 (4.370%)\n",
      "Assembled reads file...............: /code/sequences/100.assembled.fastq\n",
      "Discarded reads file...............: /code/sequences/100.discarded.fastq\n",
      "Unassembled forward reads file.....: /code/sequences/100.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /code/sequences/100.unassembled.reverse.fastq\n",
      "\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.11 [Nov 5, 2017]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /code/sequences/101.sickle.trimmed.R1.fastq.gz\n",
      "Reverse reads file.................: /code/sequences/101.sickle.trimmed.R2.fastq.gz\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: NO\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 50\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 0\n",
      "Minimum read size after trimming...: 1\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 130\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Assemblying reads: 0%\n",
      "Assemblying reads: 86%\n",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 93,131 / 100,405 (92.755%)\n",
      "Discarded reads ...................: 0 / 100,405 (0.000%)\n",
      "Not assembled reads ...............: 7,274 / 100,405 (7.245%)\n",
      "Assembled reads file...............: /code/sequences/101.assembled.fastq\n",
      "Discarded reads file...............: /code/sequences/101.discarded.fastq\n",
      "Unassembled forward reads file.....: /code/sequences/101.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /code/sequences/101.unassembled.reverse.fastq\n",
      "\n",
      "\n",
      "FastQ records kept: 84186\n",
      "FastQ records discarded: 8945\n",
      "\n",
      "\n",
      "FastQ records kept: 119431\n",
      "FastQ records discarded: 10970\n",
      "\n",
      "\n",
      "FastQ records kept: 152834\n",
      "FastQ records discarded: 13475\n",
      "\n",
      "\n",
      "Histogram /code/sequences/readyForQiime.allsamples.fasta.histogram.svg finished\n",
      "Histogram /code/sequences/102.passedQC.fastq.histogram.svg finished\n",
      "Histogram /code/sequences/100.passedQC.fastq.histogram.svg finished\n",
      "Histogram /code/sequences/101.passedQC.fastq.histogram.svg finished\n",
      "FINISHED Amplicon quality control!\n"
     ]
    }
   ],
   "source": [
    "!python ./ampliconQC.py --data sequences --forward ATGCGTTGGAGAGARCGTTTC --reverse GATCACCTTCTAATTTACCWACAACTG --threads 130 --histograms --qiime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "!for file in sequences/*.passedQC.fastq; \\\n",
    "do \\\n",
    "  awk 'NR%4==2{sum+=1}END{print FILENAME,sum}' $file >> sequences/diatomSequenceCounts.txt; \\\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sequences/100.passedQC.fastq 119431\r\n",
      "sequences/101.passedQC.fastq 84186\r\n",
      "sequences/102.passedQC.fastq 152834\r\n"
     ]
    }
   ],
   "source": [
    "!cat sequences/diatomSequenceCounts.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Assign similar sequences to OTUs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/miniconda/lib/python2.7/site-packages/matplotlib/font_manager.py:273: UserWarning: Matplotlib is building the font cache using fc-list. This may take a moment.\r\n",
      "  warnings.warn('Matplotlib is building the font cache using fc-list. This may take a moment.')\r\n"
     ]
    }
   ],
   "source": [
    "!pick_otus.py -i sequences/readyForQiime.allsamples.fasta -o sequences/picked_otus_97"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "####  Pick a representative set of sequences. For each OTU, one sequence will be used in subsequent analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "!pick_rep_set.py -i sequences/picked_otus_97/readyForQiime.allsamples_otus.txt \\\n",
    "  -f sequences/readyForQiime.allsamples.fasta \\\n",
    "  -o sequences/repset.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "!blastn -db diatoms -query sequences/repset.fasta \\\n",
    "  -out sequences/repset.diatoms.blastn \\\n",
    "  -task blastn -max_target_seqs 1 -num_threads 130 -outfmt 6 -evalue 0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir sequences/assigned_taxonomy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "!python ./create_taxonomy_assignments_from_blast.py --taxonomy diatoms.taxonomy.FINAL2017.txt \\\n",
    "  --percid 95.0 --blast sequences/repset.diatoms.blastn --output sequences/assigned_taxonomy/repset.taxonomy.txt "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "!make_otu_table.py -i sequences/picked_otus_97/readyForQiime.allsamples_otus.txt \\\n",
    "  -t sequences/assigned_taxonomy/repset.taxonomy.txt \\\n",
    "  -o sequences/otu_table.biom"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Lookup taxa from OTU table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "!filter_taxa_from_otu_table.py -i sequences/otu_table.biom \\\n",
    "  -o sequences/otu_table.diatomsonly.biom \\\n",
    "  -n MARINE,NOT_DIATOM,Yellow_green_Algae,None"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Sort OTU table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "!sort_otu_table.py -i sequences/otu_table.diatomsonly.biom \\\n",
    "  -o sequences/otu_table.diatomsonly.biom"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Summerise taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "!summarize_taxa.py -L 1 \\\n",
    "  -i sequences/otu_table.diatomsonly.biom \\\n",
    "  -o sequences/visualised_taxonomy -a"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Produce Diatom reports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reports completed\r\n"
     ]
    }
   ],
   "source": [
    "!python ./produceDiatomReports.py --folder sequences --lookup lookuptable.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Taxonomy</th>\n",
       "      <th>100763785</th>\n",
       "      <th>none present</th>\n",
       "      <th>100763824</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Taxonomy, 100763785, none present, 100763824]\n",
       "Index: []"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv('sequences/Abundances.fail.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Taxonomy</th>\n",
       "      <th>100763785</th>\n",
       "      <th>none present</th>\n",
       "      <th>100763824</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Taxonomy, 100763785, none present, 100763824]\n",
       "Index: []"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv('sequences/Abundances.pass.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
